### PR TITLE
Travis to GitHub actions

### LINKED ISSUE #1668: PRIO: Move to GitHub actions
Since Travis is not supporting FOSS anymore really, we would need to use
now GitHub Actions for example.

Example of PHP with sqlite, mysql, postgres and prefer lowest:
https://github.com/FriendsOfCake/search/pull/299

Anyone wants to try doing this?
That would be awesome.

We can also do only parts of the current test matrix and incrementally improve it.

### PR DESCRIPTION
**Pull request by stereomon**
Ref: #1682

I was able to fix the remaining issue, thanks to @cristianoc72 

I also added https://codecov.io/

This PR closes #1668

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,407 +0,0 @@
...
-        - COMPOSER=./tests/composer/composer-symfony5-max.json composer install
-        - composer show
-
-  fast_finish: true
-
-cache:
-  directories:
-    - $HOME/.composer/cache
-
-notifications:
```
Comment (stereomon) [inline .travis.yml:404]: What about those notifications, should we also add them to GA?

Comment (dereuromark) [reply .travis.yml:404]: probably not for now.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -5,6 +5,110 @@ on: [
...
+                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+                  restore-keys: ${{ runner.os }}-composer-
+
+            - name: Move specific composer.json (Symfony version ${{ matrix.symfony-version }})
+              run: mv tests/composer/composer-symfony${{ matrix.symfony-version }}.json composer.json
+
+            - name: Composer install (Symfony version ${{ matrix.symfony-version }})
+              run: composer install --no-progress --prefer-dist --optimize-autoloader
+
+            - name: Setup Postgresql dabase for test suite
```
Comment (cristianoc72) [inline .github/workflows/ci.yml:73]: `database`

This error is mine 😄

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,6 +5,110 @@ on: [
...
+              if: matrix.db-type == 'pgsql'
+              run: tests/bin/setup.pgsql.sh
+              env:
+                  PGPASSWORD: 'postgres'
+                  DB_NAME: 'propel-tests'
+                  DB_USER: 'postgres'
+                  DB_PW: 'postgres'
+
+            - name: Setup the dabase for test suite
```
Comment (dereuromark) [inline .github/workflows/ci.yml:82]: Same etc

**Review by dereuromark — APPROVED** [review]

**Review by cristianoc72 — APPROVED** [review]

**Review by cristianoc72 — APPROVED** [review]

**Comment by dereuromark** [comment]
> I also added https://codecov.io/

Do we also need to switch out the badge in readme? so far it shows codeclimate or sth.

**Comment by stereomon** [comment]
Codecov badge is in, codeclimate could be removed.

**Pull request merged**